src/libstd/fs.rs

```diff
@@ -1536,6 +1529,24 @@ mod tests {
         TempDir(ret)
     }
 
+    // Several test fail on windows if the user does not have permission to create symlinks (the
+    // `SeCreateSymbolicLinkPrivilege`). Instead of disabling these test on Windows, use this
+    // function to test whether we have permission, and return otherwise. This way, we still don't
+    // run these tests most of the time, but at least we do if the user has the right permissions.
+    pub fn got_symlink_permission(tmpdir: &TempDir) -> bool {
```

Comment:
Could this function always return `true` on unix? I'm somewhat wary about us auto-ignoring tests as it may be easy for the tests to accidentally be ignored in the future.